<launch>

    <!-- run pointcloud_to_laserscan node -->
    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">

        <remap from="cloud_in" to="/proj_cloud"/>
        <remap from="scan" to="/camera_laser_scan"/>
        <rosparam>
            target_frame: base_footprint # Leave disabled to output scan in pointcloud frame
            transform_tolerance: 0.01
            min_height: -0.1
            max_height: 0.1

            angle_min: -0.785398163 # -M_PI/4
            angle_max: 0.785398163 # M_PI/4
            angle_increment: 0.004363323 # M_PI/720.0
            scan_time: 0.1
            range_min: 0.45
            range_max: 7.5
            use_inf: true
            inf_epsilon: 1.0

            # Concurrency level, affects number of pointclouds queued for processing and number of threads used
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 1
        </rosparam>

    </node>

</launch>
